src/analysis/convex/integral.lean

```diff
@@ -195,3 +212,122 @@ lemma concave_on.le_map_integral [is_probability_measure μ] {s : set E} {g : E
   ∫ x, g (f x) ∂μ ≤ g (∫ x, f x ∂μ) :=
 by simpa only [average_eq_integral]
   using hg.le_map_average hgc hsc (is_probability_measure.ne_zero μ) hfs hfi hgi
+
+/-- If `f : α → E` is an integrable function, then either it is a.e. equal to the constant
+`⨍ x, f x ∂μ` or there exists a measurable set such that `μ s ≠ 0`, `μ sᶜ ≠ 0`, and the average
+values of `f` over `s` and `sᶜ` are different. -/
+lemma measure_theory.integrable.ae_eq_const_or_exists_average_ne_compl [is_finite_measure μ]
+  {f : α → E} (hfi : integrable f μ) :
+  (f =ᵐ[μ] const α (⨍ x, f x ∂μ)) ∨ ∃ s, measurable_set s ∧ μ s ≠ 0 ∧ μ sᶜ ≠ 0 ∧
+    ⨍ x in s, f x ∂μ ≠ ⨍ x in sᶜ, f x ∂μ :=
+begin
+  refine or_iff_not_imp_right.mpr (λ H, _), push_neg at H,
+  refine hfi.ae_eq_of_forall_set_integral_eq _ _ (integrable_const _) (λ s hs hs', _), clear hs',
+  simp only [const_apply, set_integral_const],
+  by_cases h₀ : μ s = 0,
+  { rw [restrict_eq_zero.2 h₀, integral_zero_measure, h₀, ennreal.zero_to_real, zero_smul] },
+  by_cases h₀' : μ sᶜ = 0,
+  { rw ← ae_eq_univ at h₀',
+    rw [restrict_congr_set h₀', restrict_univ, measure_congr h₀', measure_smul_average] },
+  have := average_mem_open_segment_compl_self hs.null_measurable_set h₀ h₀' hfi,
+  rw [← H s hs h₀ h₀', open_segment_same, mem_singleton_iff] at this,
+  rw [this, measure_smul_set_average _ (measure_ne_top μ _)]
+end
+
+/-- **Jensen's inequality**, strict version: if an integrable function `f : α → E` takes values in a
```

Comment:
I wouldn't mention the name `Jensen's inequality` in this docstring: for me, Jensen's inequality is saying that a convex function applied to an integral is smaller than the integral of the convex function applied. Here, you are not applying a function to an integral.